smolyay/benchmark/benchmark.py

```diff
@@ -28,19 +24,19 @@ def __call__(self,x):
             If the input is outside the function domain.
         """
         x = numpy.array(x, copy=False,ndmin=2)
-        if self.dimension > 1:
-            oob = any(numpy.any(xi < bound[0]) or numpy.any(xi > bound[1]) 
-                    for xi,bound in zip(x.transpose(),self.domain))
-        else:
-            oob = (numpy.any(x < self.domain[0][0]) or 
-                    numpy.any(x > self.domain[0][1]))
+        if self.dimension == 1 and x.shape[-1] > 1:
+            x = numpy.reshape(x,list(x.shape)+[1])
+
+        oob = any(numpy.any(x[..., i] < self.domain[i][0]) or 
+                  numpy.any(x[..., i] > self.domain[i][1]) 
+                  for i in range(self.dimension))
         if oob:
             raise ValueError("Input out domain of function.")
```

Comment:
This can go together now:
```suggestion
        if any(numpy.any(x[..., i] < self.domain[i][0]) or 
                  numpy.any(x[..., i] > self.domain[i][1]) 
                  for i in range(self.dimension)):
            raise ValueError("Input out domain of function.")
```